java-checks/src/main/java/org/sonar/java/checks/ProtectedMemberInFinalClassCheck.java

```diff
@@ -84,7 +83,8 @@ private void checkComplianceOnModifiersAndSymbol(ModifiersTree modifiers, Symbol
   }
 
   private static boolean isVisibleForTesting(Symbol symbol) {
-    return symbol.metadata().isAnnotatedWith(GUAVA_FQCN);
+    return symbol.metadata().annotations().stream()
+      .anyMatch(annotation -> "VisibleForTesting".equals(annotation.symbol().type().name()));
```

Comment:
I'm a bit afraid here of cases where the annotation is called "VisibleForTesting" but not resolved by our semantic engine. In the following example, their is no change we can resolve the type, and still, it's a VisibleForTesting annotation, so I would expect no issue.
```
@org.foo.bar.VisibleForTesting
final int var2;
```